src/App.vue

```diff
@@ -37,71 +72,112 @@ export default {
+        case "auto light":
+          document.documentElement.classList.add("auto light");
+
         default:
-          document.documentElement.classList.add('dark')
-          break
+          document.documentElement.classList.add("dark");
+          break;
       }
     },
+    updateTheme(){
+      document.documentElement.classList.remove("auto", "dark", "light", "october");
+
+      switch (this.theme) {
+        case "dark":
+          document.documentElement.classList.add("dark");
+          break;
+        case "light":
+          document.documentElement.classList.add("light");
+          break;
+        case "october":
+          document.documentElement.classList.add("dark","october");
+          break;
+        case "auto":
+          const isDark = window.matchMedia("(prefers-color-scheme: dark)").matches
+          const preferredTheme = isDark ? "dark" : "light"
+
+          document.documentElement.classList.add("auto", preferredTheme);
+          break;
+        default:
+          document.documentElement.classList.add("dark");
+          break;
+      }
+    }
   },
+
   computed: {
     home() {
-      return this.$route.name === 'home'
+      return this.$route.name === "home";
     },
     ligatures() {
-      return this.$store.state.settings.editor.ligatures
+      return this.$store.state.settings.editor.ligatures;
     },
     publicDoc() {
-      return this.$route.name === 'public_doc'
+      return this.$route.name === "public_doc";
     },
     showStripeModal() {
-      return this.$store.state.showStripeModal
+      return this.$store.state.showStripeModal;
     },
     sizes() {
-      if (this.mq.current === 'xs') return ['xs xs-plus']
-      if (this.mq.current === 'sm') return ['sm xs-plus sm-plus']
-      if (this.mq.current === 'md') return ['md xs-plus sm-plus md-plus']
-      if (this.mq.current === 'lg') return ['lg xs-plus sm-plus md-plus lg-plus']
-      if (this.mq.current === 'xl') return ['xl xs-plus sm-plus md-plus lg-plus xl-plus']
+      if (this.mq.current === "xs") return ["xs xs-plus"];
+      if (this.mq.current === "sm") return ["sm xs-plus sm-plus"];
+      if (this.mq.current === "md") return ["md xs-plus sm-plus md-plus"];
+      if (this.mq.current === "lg") return ["lg xs-plus sm-plus md-plus lg-plus"];
+      if (this.mq.current === "xl") return ["xl xs-plus sm-plus md-plus lg-plus xl-plus"];
 
-      return []
+      return [];
     },
     theme() {
-      return this.$store.state.settings.theme
+      return this.$store.state.settings.theme;
     },
+    // theme(_value){
+    //   this.updateTheme()
+    // },
+   
   },
   methods: {
     closeChangelog() {
-      this.showChangelog = false
+      this.showChangelog = false;
     },
     hideModal() {
-      this.showModal = false
+      this.showModal = false;
     },
     refreshPage() {
-      window.location.reload(true)
+      window.location.reload(true);
     },
+    theme(_value){
+      this.updateTheme()
+    }
```

Comment:
You don't need to add a new `theme` function here. You can just replace the contents of the `theme` function that's defined up in the `watch` block.